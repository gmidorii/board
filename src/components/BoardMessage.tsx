import * as React from 'react'

export interface Props {
  onClick(message: string): void
  onClear(): void
}
export interface State {}

export class BoardMessage extends React.Component<Props, State> {
  private inputId = 'messageInput'
  private submitId = 'messageSubmit'
  private clearId = 'messageClear'
  private messagePlaceHolder = 'Ctrl + Enter => submit'

  constructor(props: Props) {
    super(props)
  }

  extractMessage(): string {
    const elm: HTMLInputElement = document.getElementById(
      this.inputId
    ) as HTMLInputElement
    return elm.value
  }

  clearInput(): void {
    const elm: HTMLInputElement = document.getElementById(
      this.inputId
    ) as HTMLInputElement
    elm.value = ''
  }

  isCtrlorMetaEnter(event: React.KeyboardEvent<HTMLTextAreaElement>): boolean {
    return event.keyCode !== 13 || (!event.metaKey && !event.ctrlKey)
  }

  render() {
    return (
      <div>
        <div>
          <textarea
            id={this.inputId}
            placeholder={this.messagePlaceHolder}
            onKeyDown={event => {
              if (this.isCtrlorMetaEnter(event)) {
                return
              }
              const value = this.extractMessage()
              if (value === '') {
                return
              }
              this.props.onClick(value)
              this.clearInput()
            }}
          />
        </div>
        <button
          id={this.submitId}
          onClick={() => {
            this.props.onClick(this.extractMessage())
            this.clearInput()
          }}
        >
          submit
        </button>
        <button
          id={this.clearId}
          onClick={() => {
            this.props.onClear()
            this.clearInput()
          }}
        >
          clear
        </button>
      </div>
    )
  }
}
