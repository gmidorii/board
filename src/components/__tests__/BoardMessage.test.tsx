import * as React from 'react'
import * as render from 'react-test-renderer'
import { BoardMessage } from '../BoardMessage'

test('message snapshort', () => {
  const com = render.create(
    <BoardMessage
      onClick={() => {
        return 'hoge'
      }}
      onClear={() => {
        return 'hoge'
      }}
    />
  )
  expect(com.toJSON()).toMatchSnapshot()
})
