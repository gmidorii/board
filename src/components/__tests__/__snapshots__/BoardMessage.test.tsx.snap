// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message snapshort 1`] = `
<div>
  <div>
    <textarea
      id="messageInput"
      onKeyDown={[Function]}
      placeholder="Ctrl + Enter => submit"
    />
  </div>
  <button
    id="messageSubmit"
    onClick={[Function]}
  >
    submit
  </button>
  <button
    id="messageClear"
    onClick={[Function]}
  >
    clear
  </button>
</div>
`;
